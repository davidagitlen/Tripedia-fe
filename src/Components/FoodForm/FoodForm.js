import React, { useState } from 'react';
import './FoodForm.scss';

const FoodForm = ({ collapseForm, openForm, defaultForm }) => {
  const [form, toggleClicked] = useState({
    Mexican: false,
    Thai: false,
    Chinese: false,
    Japanese: false,
    Vietnamese: false,
    FastFood: false,
    FineDining: false,
    Pizza: false
  });

  let mockProps = [
    'Mexican',
    'Thai',
    'Chinese',
    'Japanese',
    'Vietnamese',
    'Fast Food',
    'Fine Dining',
    'Pizza'
  ];

  const checkBoxes = mockProps.map(checkBox => {
    let name = checkBox.replace(/ /gi, '');
    return (
      <div
        key={name}
        className='individual-checkbox__container'
        >
        <input
          className='checkbox'
          type='checkbox'
          value={name}
          checked={form[name]}
          onChange={e => 
            toggleClicked({
              ...form, [e.target.value]: !form[e.target.value]
            })
          }
        />
        <label>{checkBox}</label>
      </div>
    );
  });

  if(openForm.FoodForm) {
    return (
      <div className='form__container'>
        <p>- Food -</p>
        <div className='checkbox__container'>
          {checkBoxes}
        </div>
      </div>
    );
  } else {
    return (
      <div
        onClick={e => 
          collapseForm({
            ...defaultForm,
            FoodForm:
            !openForm.FoodForm
          })
        }
        className='form-closed__container'
      >
        <p>- Food -</p>
      </div>
    );
  }
}

export default FoodForm;