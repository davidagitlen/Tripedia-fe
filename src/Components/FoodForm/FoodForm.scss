.food-form__container {
  width: 100%;
  height: 8vh;
  background: goldenrod;
}