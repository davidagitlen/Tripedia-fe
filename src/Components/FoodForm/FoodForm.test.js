import React, { createContext } from 'react';
import { shallow, mount } from 'enzyme';
import FoodForm from './FoodForm';

describe('FoodForm', () => {

  it('should mock formContext and match a snapshot', () => {
    const contextValues = {
      origin: "",
      destination: "",
      tripSubmitted: false,
      attractions: {},
      accommodations: {},
      food: {},
      drinks: {},
      services: {},
      miscellaneous: {},
      selectedCategories: []
    };
    const FormContext = createContext(contextValues);
    const mockOpenForm = jest.fn();
    const mockCollapseForm = jest.fn();
    const mockDefaultForm = {
      StartForm: false,
      AttractionsForm: false,
      AccommodationsForm: false,
      FoodForm: false,
      DrinksForm: false,
      ServicesForm: false,
      SubmitTripForm: false
    };
    const mockFormObject = {
      ArtGalleries: [],
      WalkingTours: [],
      DayCamps: []
    }
    const wrapper = mount(
      <FormContext.Provider value={contextValues}>
        <FoodForm
          collapseForm={mockCollapseForm}
          openForm={mockOpenForm}
          defaultForm={mockDefaultForm}
          formObject={mockFormObject}
        />
      </FormContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});