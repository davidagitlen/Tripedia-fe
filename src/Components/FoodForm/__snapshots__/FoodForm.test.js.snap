// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodForm should mock formContext and match a snapshot 1`] = `
<FoodForm
  collapseForm={[MockFunction]}
  defaultForm={
    Object {
      "AccommodationsForm": false,
      "AttractionsForm": false,
      "DrinksForm": false,
      "FoodForm": false,
      "ServicesForm": false,
      "StartForm": false,
      "SubmitTripForm": false,
    }
  }
  formObject={
    Object {
      "ArtGalleries": Array [],
      "DayCamps": Array [],
      "WalkingTours": Array [],
    }
  }
  openForm={[MockFunction]}
>
  <div
    className="form-closed__container"
    onClick={[Function]}
  >
    <img
      alt="food"
      src="food-pin.svg"
    />
    <p>
      - Food -
    </p>
  </div>
</FoodForm>
`;
